packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -218,7 +223,22 @@ abstract contract PermissionManager is Initializable {
     /// @param _who The address (EOA or contract) owning the permission.
     /// @param _permissionId The permission identifier.
     function _grant(address _where, address _who, bytes32 _permissionId) internal virtual {
-        _grantWithCondition(_where, _who, _permissionId, IPermissionCondition(ALLOW_FLAG));
```

Comment:
There is not much overlap anymore.
```js
function _grant(address _where, address _who, bytes32 _permissionId) internal virtual {
    if (_where == ANY_ADDR || _who == ANY_ADDR) {
        revert AnyAddressDisallowedForWhoAndWhere();
    }
    bytes32 permHash = permissionHash(_where, _who, _permissionId);
    address currentFlag = permissionsHashed[permHash];
    // Means permHash is not currently set.
    if (currentFlag == UNSET_FLAG) {
        permissionsHashed[permHash] = ALLOW_FLAG;
        emit Granted(_permissionId, msg.sender, _where, _who, IPermissionCondition(ALLOW_FLAG));
    }
}
```

```js
function _grantWithCondition(address _where, address _who, bytes32 _permissionId, IPermissionCondition _condition
) internal virtual {
    if (!address(_condition).isContract()) {
        revert ConditionInvalid(_condition);
    }
    if (_where == ANY_ADDR && _who == ANY_ADDR) {
        revert AnyAddressDisallowedForWhoAndWhere();
    }
    if (_where == ANY_ADDR || _who == ANY_ADDR) {
        bool isRestricted = isPermissionRestrictedForAnyAddr(_permissionId);
        if (_permissionId == ROOT_PERMISSION_ID || isRestricted) {
            revert PermissionsForAnyAddressDisallowed();
        }
    }
    bytes32 permHash = permissionHash(_where, _who, _permissionId);
    address currentCondition = permissionsHashed[permHash];
    address newCondition = address(_condition);
    // Means permHash is not currently set.
    if (currentCondition == UNSET_FLAG) {
        permissionsHashed[permHash] = newCondition;
        emit Granted(_permissionId, msg.sender, _where, _who, _condition);
    } else if (currentCondition != newCondition) {
        // Revert if `permHash` is already granted, but uses a different condition.
        // If we don't revert, we either should:
        //   - allow overriding the condition on the same permission
        //     which could be confusing whoever granted the same permission first
        //   - or do nothing and succeed silently which could be confusing for the caller.
        revert PermissionAlreadyGrantedForDifferentCondition({
            where: _where,
            who: _who,
            permissionId: _permissionId,
            currentCondition: currentCondition,
            newCondition: newCondition
        });
    }
}
```

Splitting the two methods apart seems good because fundamentally different things are happening.